tools/PI/DevHome.PI/BarWindow.cs

```diff
@@ -82,7 +82,9 @@ public BarWindow()
         _verticalWindow = new BarWindowVertical(_viewModel);
 
         _horizontalWindow.Closed += Window_Closed;
+        _horizontalWindow.Closed += _verticalWindow.WindowEx_Closed;
```

Comment:
>; [](http://example.com/codeflow?start=67&length=1)

Does this actually close the windows? Maybe explicitly call _horizontalWindow.Close()/_verticalWindow.Close() in the Window_Closed handler?

Reply:
It doesn't close the windows, only stops the features started by the window.

I originally wanted to call _horizontalWindow.Close()/_verticalWindow.Close(), but this will cause infinite loop. For example:
_verticalWindow.WindowEx_Closed() -> _horizontalWindow.Close() -> _horizontalWindow.WindowEx_Closed ->_verticalWindow.Close()->_verticalWindow.WindowEx_Closed() and so on

Reply:
you would need to maintain state to prevent a loop, something like an atomic<bool> closeInProgress; if (closeInProgress) { return} else {closeotherWindow();}

Reply:
But I would echo Tim's comments if there were a way we can move off the bar window that would be the better long term solution.

Reply:
Updated the code to call barwindow Close() from both horizontal and vertical windows and preventing the loop by using a bool isclosing.